### PR TITLE
Move assignment of xchar to declaration

### PR DESCRIPTION
**Pull request by AtariDreams**
This follows best practices without changing behavior

### CODE REVIEW

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks, this is a clear improvement! I pushed a commit for an additional occurrence in the `double` counterpart.

**Review by strega-nil-ms — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

File: stl/src/xdnorm.cpp [hunk 1]
```diff
@@ -8,10 +8,9 @@
 _EXTERN_C_UNLESS_PURE
 
 short _Dnorm(_Dval* ps) { // normalize double fraction
-    short xchar;
+    short xchar         = 1;
```
Comment (CaseyCarter) [inline stl/src/xdnorm.cpp:11]: The scope is small enough here as to make no difference, but in the future I suggest moving the declaration to the initialization in preference to moving the initialization to the declaration. We generally try to limit the lifetime of objects to the portion of the code that actually uses those objects.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for this code cleanup! :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat: :smile_cat:

**Pull request merged**